### PR TITLE
Migrate Page to fela and add PageContent component

### PR DESCRIPTION
**Pull request by jculvey**
This does a few things:
  1. Migrates Page to fela
  2. Adds a new PageContent component for handling horizontal spacing

### CODE REVIEW

File: packages/cf-component-page/src/PageHeader.js [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+const PageSubheading = createComponent(
+  ({ theme }) => ({
+    ...theme.subheading
+  }),
+  'h2'
+);
+
+const PageHeader = ({ title, subtitle, className }) => {
+  return (
+    <header className={className}>
```
Comment (koddsson) [inline packages/cf-component-page/src/PageHeader.js]: Why a `className` here and not use `fela`?

Comment (tajo) [inline packages/cf-component-page/src/PageHeader.js]: `PageHeader` is wrapped by `createComponent` before exporting, so it uses fela and doesn't expose className.

File: packages/cf-component-page/src/PageHeader.js [hunk 2]
```diff
@@ -1,22 +1,41 @@
 import React from 'react';
-
 import PropTypes from 'prop-types';
+import { createComponent } from 'cf-style-container';
+
+const PageHeading = createComponent(
+  () => ({
+    fontSize: '2rem',
+    lineHeight: '1.2',
+    fontWeight: 400
```
Comment (manatarms) [inline packages/cf-component-page/src/PageHeader.js]: style-const?

File: packages/cf-component-page/src/PageHeader.js (hunk 1 above)
Comment (koddsson) [inline packages/cf-component-page/src/PageHeader.js]: Is the `className` here provided by `fela`? This is the same thing as doing:

```js
createComponent(
  () = > {},
  'header',
  ['title', 'subtitle']
);
```
?

**Pull request merged**